File: BaekJoon/ITBigTech/PronouncePWD/PWD.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strings"
)

func isDouble(input string) bool {
	for i := 0; i < len(input)-1; i++ {
		if input[i] == input[i+1] && input[i] != 'e' && input[i] != 'o' {
			return false
		}
	}
	return true
}

func main() {

	// 모음 포함 유무
	isVowel := regexp.MustCompile(`[aeiou]`)
	// 자음이나 모음 3번 연속
	isTriple := regexp.MustCompile(`[aeiou]{3}|[^aeiou]{3}`)
	// ee와 oo를 제외한 같은 글자 2번 연속
	// isDouble()

	reader := bufio.NewReader(os.Stdin)

	for {
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		if input == "end" {
			break
		}

		case1 := isVowel.MatchString(input)
		case2 := !isTriple.MatchString(input)
		case3 := isDouble(input)

		if case1 && case2 && case3 {
			fmt.Printf("<%s> is acceptable.\n", input)
		} else {
			fmt.Printf("<%s> is not acceptable.\n", input)
		}
	}
}

File: BaekJoon/ITBigTech/PronouncePWD4659/PWD.java
import java.util.Scanner;

public class PWD {

    // 모음인가?
    public static boolean isVowel(char c) {
        return "aeiou".indexOf(c) != -1;
    }

    // 모음 존재 유무
    public static boolean contiansVowel(String input) {
        for (char c : input.toCharArray()) {
            // c가 aeiou 중에 있으면 true
            if ("aeiou".indexOf(c) != -1) {
                return true;
            }
        }
        return false;
    }

    // 모음이나 자음이 연속 3번 있는지
    public static boolean isTriple(String input) {
        for (int i = 0; i < input.length() - 2; i++) {
            // 모음 연속 3번
            if (isVowel(input.charAt(i)) && isVowel(input.charAt(i+1)) && isVowel(input.charAt(i+2))) {
                return true;
            }
            // 자음 연속 3번
            if (!isVowel(input.charAt(i)) && !isVowel(input.charAt(i+1)) && !isVowel(input.charAt(i+2))) {
                return true;
            }
        }
        return false;
    }

    // ee나 oo를 제외한 글자가 연속 2번 있는지
    public static boolean isDouble(String input) {
        for (int i = 0; i < input.length() - 1; i++) {
            if (input.charAt(i) == input.charAt(i+1) && input.charAt(i) != 'e' && input.charAt(i) != 'o') {
                return true;
            }
        }
        return false;
    }

    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        
        while (true) {
            String input = sc.nextLine();
            if (input.equals("end")) break;

            boolean case1 = contiansVowel(input);
            boolean case2 = !isTriple(input);
            boolean case3 = !isDouble(input);

            if (case1 && case2 && case3) {
                System.out.println("<" + input + "> is acceptable.");
            } else {
                System.out.println("<" + input + "> is not acceptable.");
            }
        }
        sc.close();
    }
}
